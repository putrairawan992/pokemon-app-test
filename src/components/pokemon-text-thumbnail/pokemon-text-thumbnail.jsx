import React from "react";
import T from "prop-types";
import Card from "../card/card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import ButtonBase from "@material-ui/core/ButtonBase";

const propTypes = {
  text: T.string,
  onClick: T.func,
};

export const PokemonTextThumbnail = ({ text, onClick }) => {
  return (
    <ButtonBase onClick={onClick}>
      <Card>
        <CardActionArea>
          <CardContent>
            <Typography gutterBottom variant="h5">
              {text}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </ButtonBase>
  );
};

PokemonTextThumbnail.propTypes = propTypes;
