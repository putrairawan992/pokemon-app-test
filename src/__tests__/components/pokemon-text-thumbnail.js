import React from "react";
import renderer from "react-test-renderer";
import { PokemonTextThumbnail } from "../../components/pokemon-text-thumbnail/pokemon-text-thumbnail";

it("Should render pokemon text thumbnail correctly", () => {
  const handleClick = jest.fn();
  const component = renderer.create(
    <PokemonTextThumbnail text="bulbasaur" onClick={handleClick} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
