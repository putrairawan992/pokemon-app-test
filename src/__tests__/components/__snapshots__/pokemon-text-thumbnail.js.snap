// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render pokemon text thumbnail correctly 1`] = `
<button
  className="MuiButtonBase-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-4 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="MuiCardContent-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          bulbasaur
        </h5>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</button>
`;
